File: nextjs-frontend/lib/types/xray.ts
export interface XrayInbound {
  id: number;
  nodeId: number;
  tag: string;
  protocol: string;
  listen: string;
  port: number;
  settingsJson: string;
  streamSettingsJson: string | null;
  sniffingJson: string | null;
  remark: string | null;
  enable: number;
  createdTime: number;
  updatedTime: number;
}

export interface XrayClient {
  id: number;
  inboundId: number;
  userId: number;
  email: string;
  uuidOrPassword: string;
  flow: string | null;
  alterId: number;
  totalTraffic: number;
  upTraffic: number;
  downTraffic: number;
  expTime: number | null;
  enable: number;
  remark: string | null;
  createdTime: number;
  updatedTime: number;
}

export interface XrayTlsCert {
  id: number;
  nodeId: number;
  domain: string;
  publicKey: string;
  privateKey: string;
  autoRenew: number;
  expireTime: number | null;
  createdTime: number;
  updatedTime: number;
}

export type XrayProtocol = 'vmess' | 'vless' | 'trojan' | 'shadowsocks';
export type XrayNetwork = 'tcp' | 'ws' | 'grpc' | 'h2';
export type XraySecurity = 'none' | 'tls' | 'reality';

File: go-backend/dto/xray.go
package dto

type XrayInboundDto struct {
	NodeId             int64  `json:"nodeId" binding:"required"`
	Tag                string `json:"tag"`
	Protocol           string `json:"protocol" binding:"required"`
	Listen             string `json:"listen"`
	Port               int    `json:"port" binding:"required"`
	SettingsJson       string `json:"settingsJson"`
	StreamSettingsJson string `json:"streamSettingsJson"`
	SniffingJson       string `json:"sniffingJson"`
	Remark             string `json:"remark"`
	CustomEntry        string `json:"customEntry"`
}

type XrayInboundUpdateDto struct {
	ID                 int64   `json:"id" binding:"required"`
	Tag                string  `json:"tag"`
	Protocol           string  `json:"protocol"`
	Listen             string  `json:"listen"`
	Port               *int    `json:"port"`
	SettingsJson       string  `json:"settingsJson"`
	StreamSettingsJson string  `json:"streamSettingsJson"`
	SniffingJson       string  `json:"sniffingJson"`
	Remark             string  `json:"remark"`
	CustomEntry        *string `json:"customEntry"`
}

type XrayClientDto struct {
	InboundId      int64  `json:"inboundId" binding:"required"`
	UserId         int64  `json:"userId"`
	UuidOrPassword string `json:"uuidOrPassword"`
	Flow           string `json:"flow"`
	AlterId        *int   `json:"alterId"`
	TotalTraffic   *int64 `json:"totalTraffic"`
	ExpTime        *int64 `json:"expTime"`
	LimitIp        *int   `json:"limitIp"`
	Reset          *int   `json:"reset"`
	Remark         string `json:"remark"`
}

type XrayClientUpdateDto struct {
	ID             int64  `json:"id" binding:"required"`
	Email          string `json:"email"`
	UuidOrPassword string `json:"uuidOrPassword"`
	Flow           string `json:"flow"`
	AlterId      *int   `json:"alterId"`
	TotalTraffic *int64 `json:"totalTraffic"`
	ExpTime      *int64 `json:"expTime"`
	LimitIp      *int   `json:"limitIp"`
	Reset        *int   `json:"reset"`
	Enable       *int   `json:"enable"`
	Remark       string `json:"remark"`
}

type XrayTlsCertDto struct {
	NodeId        int64  `json:"nodeId" binding:"required"`
	Domain        string `json:"domain" binding:"required"`
	PublicKey     string `json:"publicKey"`
	PrivateKey    string `json:"privateKey"`
	AutoRenew     *int   `json:"autoRenew"`
	AcmeEnabled   *int   `json:"acmeEnabled"`
	AcmeEmail     string `json:"acmeEmail"`
	ChallengeType string `json:"challengeType"`
	DnsProvider   string `json:"dnsProvider"`
	DnsConfig     string `json:"dnsConfig"`
	ExpireTime    *int64 `json:"expireTime"`
}

type XrayCertIssueDto struct {
	ID int64 `json:"id" binding:"required"`
}

type XrayCertRenewDto struct {
	ID int64 `json:"id" binding:"required"`
}
